packages/runtime/runtime-utils/src/summarizerNode.ts

```diff
@@ -149,10 +149,17 @@ function decodeSummary(snapshot: ISnapshotTree, logger: Pick<ITelemetryLogger, "
                         if (outstandingOps.length > 0 && newOutstandingOps.length > 0) {
                             const latestSeq = outstandingOps[outstandingOps.length - 1].sequenceNumber;
                             const newEarliestSeq = newOutstandingOps[0].sequenceNumber;
-                            assert(
-                                latestSeq < newEarliestSeq,
-                                `latestSeq exceeds newEarliestSeq in decodeSummary: ${latestSeq} >= ${newEarliestSeq}`,
-                            );
+                            if (newEarliestSeq <= latestSeq) {
+                                logger.sendTelemetryEvent({
+                                    eventName:"DuplicateOutstandingOps",
+                                    category: "generic",
+                                    // eslint-disable-next-line max-len
+                                    message: `latestSeq exceeds newEarliestSeq in decodeSummary: ${latestSeq} >= ${newEarliestSeq}`,
+                                });
+                                while (newOutstandingOps[0].sequenceNumber <= latestSeq) {
```

Comment:
Shifting can result in empty array, Maybe it still works (due to magic of JS math on undefined), but I'd rather be explicit - if array is empty, bail out.
You can actually directly calculate the point where you need to slice an array, I believe, assuming it's continuum.

Reply:
It is also fairly easy to repro the bug on old code, so you could create a bad doc using clicker and then open it with your changes and verify things worked as expected on load. (That was how I initially tested the fix)

Otherwise you may want to do a more explicit remove duplicates fix or like Vlad said. In reality though, given how the bug works, the slice point is just remove all the ops from `newOutstandingOps`.

Reply:
wish we had automated tests here. we have similar tests in shared string with checked in snapshots

Reply:
#4019